037-kafka-canary-roadmap.md

```diff
@@ -0,0 +1,127 @@
+* Similarly with/without using consumer groups. Using different group ids to detect problems with group coordinators.
+
+
+### Actionable documentation
+
+Assuming the differential observations feature proves to provide worthwhile information, we would proceed to also provide documented procedures recommending corrective actions based on the specific faults it reported.
+
+### Changes and removals
+
+* As logging mechanism will be used Log4j2
+  * The log levels will have to be changed from `0, 1, 2` to `INFO, DEBUG, WARN, TRACE`.
+  * `VERBOSITY_LOG_LEVEL` will be changed to `CANARY_LOG_LEVEL` and used inside the `log4j2.properties` file - default will be `INFO`
+* `SARAMA_LOG_ENABLED` environment variable will be removed
+* Implementation of dynamic watcher will be removed 
+  * For logging changes the dynamic reconfiguration will be used from Log4j2 (similarly to operators repository)
+
+## Affected/not affected projects
+
+This affects the Canary project only.
+
+## Compatibility
+
+The API of the existing Canary comprises the following:
+
+* The configuration env vars
+* Metrics (via the Prometheus scrape endpoint)
+* The status and health endpoints
+
+### Configuration
+
+Java clients are normally configured via `.properties` files.
+Providing a compatibility mapping from env var to properties would be a lot of work, given that the Canary hasn't seen very wide usage.
+Some options would become obselete (e.g. `KAFKA_VERSION`), others are incompatible with how the Java clients work (e.g. `KAFKA_BOOTSTRAP_BACKOFF_MAX_ATTEMPTS`).
+While we can strive to minimize incompatibilities, it seems of questionable value.
+So it seems better to provide a clean break.
+
+### Metrics
+
+It seems likely that most of the [metric names](https://github.com/strimzi/strimzi-canary#metrics-1) and labels can be maintained.
+
+There are metrics which will be deleted or changed:
+
+* Metrics that will be removed:
+  * `topic_alter_assignments_error_total` - Canary in Java will not have dynamic reassignment
+  * `consumer_timeout_join_group_total` - connected to Sarama client, Java clients don't have such error
+
+* Metrics that will be changed:
+  * `topic_describe_cluster_error_total` - will be changed to `describe_cluster_error_total`
+
+* Metrics that need to be re-evaluated:
+  * `producer_refresh_metadata_error_total` and `consumer_refresh_metadata_error_total`
+    * with Sarama, there is a possibility to manually refresh metadata of both producer and consumer
+    * there is no such possibility in Java
+
+Detailed assessment of whether the metric values were being measured in a compatible way is beyond the scope of this document.
+If measurements were incompatible, it would require existing users to adjust the thresholds used in any alerting they might have that was based on the old metrics.
+
+### Status and Health endpoints
+
+Providing compatibility for [these](https://github.com/strimzi/strimzi-canary#endpoints) should be trivial.
```

Comment:
You changed so many things in a backwards incompatible manner that this might not matter anymore.